spec/datadog/appsec/actions_handler_spec.rb

```diff
@@ -97,4 +101,124 @@
       end
     end
   end
+
+  describe '.generate_stack' do
```

Comment:
TBH this test suite look overly complicated and violates certain test practices and it would be easier to go over in 1:1 instead of chatting here.